adapters/verizonmedia/verizonmedia.go

```diff
@@ -0,0 +1,181 @@
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type VerizonMediaAdapter struct {
+	http *adapters.HTTPAdapter
+	URI  string
+}
+
+func (a *VerizonMediaAdapter) Name() string {
+	return "verizonmedia"
+}
+
+func (a *VerizonMediaAdapter) SkipNoCookies() bool {
+	return false
+}
+
+func (a *VerizonMediaAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	errors := make([]error, 0, 1)
+	if len(request.Imp) == 0 {
+		err := &errortypes.BadInput{
+			Message: "No impression in the bid request",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	var bidderExt adapters.ExtImpBidder
+	err := json.Unmarshal(request.Imp[0].Ext, &bidderExt)
+	if err != nil {
+		err = &errortypes.BadInput{
+			Message: "ext.bidder not provided",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+	var verizonMediaExt openrtb_ext.ExtImpVerizonMedia
+	err = json.Unmarshal(bidderExt.Bidder, &verizonMediaExt)
+	if err != nil {
+		err = &errortypes.BadInput{
+			Message: err.Error(),
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	if verizonMediaExt.Dcn == "" {
+		err = &errortypes.BadInput{
+			Message: "Missing param dcn",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	if verizonMediaExt.Pos == "" {
+		err = &errortypes.BadInput{
+			Message: "Missing param pos",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	siteCopy := *request.Site
+	request.Site = &siteCopy
+	changeRequestForBidService(request, &verizonMediaExt)
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	thisURI := a.URI
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Set("User-Agent", request.Device.UA)
+	headers.Add("x-openrtb-version", "2.5")
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     thisURI,
+		Body:    reqJSON,
+		Headers: headers,
+	}}, errors
+}
+
+func (a *VerizonMediaAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d.", response.StatusCode),
+		}}
+	}
+
+	var bidResp openrtb.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Bad server response: %d.", err),
+		}}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(internalRequest.Imp))
+
```

Comment:
Some other adapters throw an error if the `SeatBid` array is empty. Is this desirable in our code?
```
--- a/adapters/verizonmedia/verizonmedia.go
+++ b/adapters/verizonmedia/verizonmedia.go
@@ -115,6 +115,11 @@ func (a *VerizonMediaAdapter) MakeBids(internalRequest *openrtb.BidRequest, exte

        bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(internalRequest.Imp))

+       if len(bidResp.SeatBid) < 1 {
+               return nil, []error{
+                       newBadServerResponseError(fmt.Sprintf("Invalid SeatBids count: %d", len(bidResp.SeatBid))),
+               }
+       }
        for _, sb := range bidResp.SeatBid {
                for _, bid := range sb.Bid {
                        exists, mediaTypeId := getImpInfo(bid.ImpID, internalRequest.Imp)
```

Reply:
Let me ask. Few months ago I was told that for a 200 response we'll always get an ad, even if is the default ad.

Reply:
The new API responds with an empty "seatbid" array in case the inventory is not found.

Reply:
We added this extra check. Even if the exception message is not really helpful, empty seadbid could happen because of wrong values for the ext params this way is more visible that our bank-end could not handled it.